rdt/hyper_transformer.py

```diff
@@ -798,24 +798,30 @@ def _reverse_transform(self, data, prevent_subset):
             raise NotFittedError(self._NOT_FIT_MESSAGE)
 
         unknown_columns = self._subset(data.columns, self._output_columns, not_in=True)
+        if unknown_columns:
+            raise Error(
+                'There are unexpected column names in the data you are trying to transform. '
+                f'A reverse transform is not defined for {unknown_columns}.'
+            )
+
         if prevent_subset:
             contained = all(column in self._output_columns for column in data.columns)
             is_subset = contained and len(data.columns) < len(self._output_columns)
-            if unknown_columns or is_subset:
+            if is_subset:
                 raise Error(
                     'There are unexpected columns in the data you are trying to transform. '
                     'You must provide a transformed dataset with all the columns from the '
                     'original data.'
                 )
 
-        elif unknown_columns:
-            raise Error(
-                'There are unexpected column names in the data you are trying to transform. '
-                f'A reverse transform is not defined for {unknown_columns}.'
-            )
+            for transformer in reversed(self._transformers_sequence):
+                data = transformer.reverse_transform(data, drop=False)
 
-        for transformer in reversed(self._transformers_sequence):
-            data = transformer.reverse_transform(data, drop=False)
+        else:
+            for transformer in reversed(self._transformers_sequence):
+                output_columns = transformer.get_output_columns()
+                if output_columns and set(output_columns).issubset(data.columns):
```

Comment:
Why is it necessary to check that output_columns is not an empty list? Instead of just:
`if set(output_columns).issubset(data.columns):`

Reply:
Because some transformers, generators, return an empty list, and that conditions ends up being `True` if so:
```python
In [1]: set(list()).issubset(['a'])
Out[1]: True
```